packages/react/src/components/Embed/Embed.tsx

```diff
@@ -142,25 +151,27 @@ class Embed extends AutoControlledComponent<WithAsProp<EmbedProps>, EmbedState>
                 controls: false,
                 loop: true,
                 muted: true,
+                poster: placeholder,
                 styles: styles.video,
                 variables: {
                   width: variables.width,
                   height: variables.height,
                 },
               },
             })}
-            {Box.create(iframe, { defaultProps: { as: 'iframe', styles: styles.iframe } })}
+            {Box.create(iframe, {
+              defaultProps: {
+                as: 'iframe',
+                styles: styles.iframe,
+                style: { visibility: iframeLoaded ? 'visible' : 'hidden' },
+                onLoad: () => {
+                  this.setState({ iframeLoaded: true })
```

Comment:
Should be handled in `overrideProps` otherwise `onLoad` passed by user will override our handler.

Reply:
1. Original handler is still not called.
2. Please move it to class method:

```tsx
handleFrameOverrides = (predefinedProps) => ({
  onLoad: (e: EVENT_TYPE) => {
     _.invoke(predefinedProps, 'onLoaded', e)
     this.setState({ iframeLoaded: true })
  }
}}
```